frontend/model/contracts/group.js

```diff
@@ -63,6 +69,67 @@ function initGroupProfile (contractID: string) {
   }
 }
 
+export function removeMemberSecret () {
+  return {
+    secret: `${parseInt(Math.random() * 10000)}` // TODO: this
+  }
+}
+
+export function validateRemoveMember (groupState, { data, meta }) {
+  const memberToRemove = data.member
+  const membersCount = Object.keys(groupState.profiles).length
+
+  if (!groupState.profiles[memberToRemove]) {
+    console.error(`validateRemoveMember - ${memberToRemove} isn't part of the group`)
+    throw new GIErrorUIRuntimeError(L('The member {memberToRemove} is not part of the group.', { memberToRemove }))
+  }
+
+  if (membersCount === 1) {
+    console.error('validateRemoveMember - only 1 member')
+    throw new GIErrorUIRuntimeError(L('The group only has one member and you cannot remove yourself.'))
+  }
+
+  if (membersCount === 2) {
+    if (meta.username !== groupState.settings.groupCreator) {
+      console.error('validateRemoveMember - not the group creator')
+      throw new GIErrorUIRuntimeError(L('In a group of 2, only the group creator can remove the other member.'))
+    }
+  } else {
+    const { payload } = groupState.proposals[data.proposalHash] || {}
+    if (payload && payload.secret === data.proposalPayload) {
+      console.error('validateRemoveMember - invalid associated proposal')
+      throw new GIErrorUIRuntimeError(L('The proposal associated to remove {memberToRemove} is not valid.', { memberToRemove }))
```

Comment:
Note that `GIErrorUIRuntimeError` must never be called anywhere in this file because those errors are specifically for errors that happen via Vue.js code that is triggered by a user action (e.g. clicking a button).

During processing of messages, only one of `GIErrorIgnoreAndBanIfGroup`, `GIErrorDropAndReprocess`, or `GIErrorUnrecoverable` may be thrown (or `TypeError`... see #694).

Validation errors should throw `TypeError`, so all of the above should be `throw new TypeError`.